src/ape/contracts/base.py

```diff
@@ -48,6 +48,15 @@ def __init__(
     def __repr__(self) -> str:
         return self.abi.signature if self.abi else "constructor()"
 
+    def encode_input(self, *args) -> HexBytes:
```

Comment:
Curious if these all might be better named `encode/decode_calldata` instead of `*_input`

Reply:
I did that first, but I found that `input` / `output` describes the high-level nature of these methods for aptly whereas `calldata` and `returndata` describe the low level data nicely. In `EcoystemAPI`, we use `calldata` and `returndata` this way.

Also, this way it is the same as Brownie.

But I don't feel super strongly about it, just was explaining my thought process for why I thought `input` made more sense.

Reply:
Okay that's fair